interfaces/cython/cantera/func1.pyx

```diff
@@ -46,36 +47,110 @@ cdef class Func1:
         >>> f3(6)
         30.0
 
-    For simplicity, constant functions can be defined by passing the constant
+    For simplicity, constant functions can be defined by passing a constant
     value directly::
 
         >>> f4 = Func1(2.5)
         >>> f4(0.1)
         2.5
 
+    A `Func1` object representing a tabulated function is defined by sample
+    points and corresponding function values. Inputs are specified either by
+    two iterable objects containing sample point location and function values,
+    or a single array that concatenates those inputs in two columns. Between
+    sample points, values are evaluated based on the keyword argument
+    'interpolation'; options are 'linear' (linear interpolation, default) or
+    'previous' (nearest previous value). Outside the sample interval, the value
+    at the closest end point is returned.
+
+    Examples for tabulated `Func1` object are::
+
+        >>> t1 = Func1([[0, 2], [1, 1], [2, 0]])
+        >>> [t1(v) for v in [-0.5, 0, 0.5, 1.5, 2, 2.5]]
+        [2.0, 2.0, 1.5, 0.5, 0.0, 0.0]
```

Comment:
I think it might make more sense to introduce a new `TabulatedFunc1` class in Python rather than overloading the constructor for `Func1` based on argument types. That would leave the door open for introducing other function types, and if I'm following this implementation correctly, remove the need to expose the `Const1` class.

Reply:
I exposed the `Const1` class, as the C++ implementation is presumably more efficient (and it was already implemented).

Further, the reason for keeping everything in a single constructor was inspired by the pre-existing exemption of constants (which are now handled by the C++ layer). I'd either change both (defining a new `Const1` class, and deprecating the old behavior), or leave things as they are.

As an aside: if we start to introduce derived classes, I may add a `Func1` derived function implementing a crank function, as this is frequently needed for engine-related applications.

Reply:
I guess my point is that currently, `Func1` objects in Python are almost always created automatically as a wrapper when the user provides either a Python function object or a constant to a method like `Wall.set_velocity`. With the tabulated formulation, I would expect users to write something more like:
```python
wall.set_velocity(TabulatedFunc([0,1,2], [0.1, 0.2, 0.3]))
```
rather than simply:
```py
wall.set_velocity([[0,1,2], [0.1, 0.2, 0.3]])
```
I'm fine with splitting out `Const1` as long as the existing behavior for functions like `Wall.set_velocity` is preserved.

Reply:
Makes sense. Hadn't really considered this auto-generated wrapper case.

Reply:
Alright. The `Tabulated1` Python class is now introduced (nomenclature matches C++), i.e.

```python
wall.set_velocity(Tabulated1([0,1,2], [0.1, 0.2, 0.3]))
```

After running some tests, it turns out the the C++ version of `Const1` is actually 50% slower (!) than the Python `lambda` function (likely Cython overhead?); accordingly, there's no point in exposing and/or introducing an interface for C++ `Const1`.